pkg/cmd/extension/manager_test.go

```diff
@@ -506,6 +506,71 @@ func TestManager_UpgradeExtension_BinaryExtension(t *testing.T) {
 	assert.Equal(t, "", stderr.String())
 }
 
+func TestManager_UpgradeExtension_BinaryExtension_Pinned_Force(t *testing.T) {
```

Comment:
this seemed the best test to duplicate, changing the pinned = true in `m.upgradeExtension(ext, true)`, rather than modify  `TestManager_UpgradeExtension_BinaryExtension` to have it implicitly test pinned = false